ppb_vector/vector2.py

```diff
@@ -115,6 +134,7 @@ def reflect(self, surface_normal: 'Vector2') -> 'Vector2':
         """
         Calculate the reflection of the vector against a given surface normal
         """
+        surface_normal = _mkvector(surface_normal)
```

Comment:
If we do this, we need to change the type hint.

On that front, we should probably build a union or type var for this.